src/pages/Login/LoginState.js

```diff
@@ -0,0 +1,265 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { FaInfoCircle } from 'react-icons/fa';
+import { GoPasskeyFill } from 'react-icons/go';
+import { Trans, useTranslation } from 'react-i18next';
+import { CSSTransition } from 'react-transition-group';
+import OnlineStatusContext from '../../context/OnlineStatusContext';
+import { useApi } from '../../api';
+import { useLocalStorageKeystore } from '../../services/LocalStorageKeystore';
+import logo from '../../assets/images/logo.png';
+import GetButton from '../../components/Buttons/GetButton';
+import { PiWifiHighBold, PiWifiSlashBold } from "react-icons/pi";
+
+
+const WebauthnSignupLogin = ({
+	isLogin,
+	isSubmitting,
+	setIsSubmitting,
+	filteredUser,
+}) => {
+	const { isOnline } = useContext(OnlineStatusContext);
+
+	const api = useApi(isOnline);
+	const [inProgress, setInProgress] = useState(false);
+	const [error, setError] = useState('');
+	const [needPrfRetry, setNeedPrfRetry] = useState(false);
+	const [resolvePrfRetryPrompt, setResolvePrfRetryPrompt] = useState(null);
+	const [prfRetryAccepted, setPrfRetryAccepted] = useState(false);
+	const navigate = useNavigate();
+	const location = useLocation();
+	const from = location.state?.from || '/';
+
+	const { t } = useTranslation();
+	const keystore = useLocalStorageKeystore();
+	useEffect(
+		() => {
+			setError("");
+		},
+		[isLogin],
+	);
+
+	const promptForPrfRetry = async () => {
+		setNeedPrfRetry(true);
+		return new Promise((resolve, reject) => {
+			setResolvePrfRetryPrompt(() => resolve);
+		}).finally(() => {
+			setNeedPrfRetry(false);
+			setPrfRetryAccepted(true);
+			setResolvePrfRetryPrompt(null);
+		});
+	};
+
+	const onLogin = useCallback(
+		async (cachedUser) => {
+			const result = await api.loginWebauthn(keystore, promptForPrfRetry, cachedUser);
+			if (result.ok) {
+
+				navigate(from, { replace: true });
+
+			} else {
+				// Using a switch here so the t() argument can be a literal, to ease searching
+				switch (result.val) {
+					case 'loginKeystoreFailed':
+						setError(t('loginSignup.loginKeystoreFailed'));
+						break;
+
+					case 'passkeyInvalid':
+						setError(t('loginSignup.passkeyInvalid'));
+						break;
+
+					case 'passkeyLoginFailedTryAgain':
+						setError(t('loginSignup.passkeyLoginFailedTryAgain'));
+						break;
+
+					case 'passkeyLoginFailedServerError':
+						setError(t('loginSignup.passkeyLoginFailedServerError'));
+						break;
+
+					default:
+						throw result;
+				}
+			}
+		},
+		[api, keystore, navigate, t],
+	);
+
+	const onSubmit = async (event) => {
+		event.preventDefault();
+
+		setError();
+		setInProgress(true);
+		setIsSubmitting(true);
+
+		await onLogin();
+
+		setInProgress(false);
+		setIsSubmitting(false);
+	};
```

Comment:
I don't think there's any way to trigger this callback, is there? The form doesn't contain any `type="submit"` as far as I can tell. So I think this callback is unused?